yaziPlugins: update on 2025-06-06

khaneliman opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
Closes https://github.com/NixOS/nixpkgs/pull/415073
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

khaneliman committed:
yaziPlugins.update: allow updating all in set

Signed-off-by: Austin Horstman <khaneliman12@gmail.com>

khaneliman committed:
yaziPlugins.update: allow committing changes

Signed-off-by: Austin Horstman <khaneliman12@gmail.com>

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414418`
Commit: `b5f8db9312b579e5fb599fa063fb526d58f0fc4f`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.restore</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.restore</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.restore</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.restore</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

GaetanLepage commented on pkgs/by-name/ya/yazi/plugins/update.py:
```diff
@@ -17,16 +19,17 @@ def run_command(cmd: str, capture_output: bool = True) -> str:
     """Get plugin repository information from Nix"""
-    owner = run_command(f"nix eval --raw -f {nixpkgs_dir} yaziPlugins.\"{plugin_name}\".src.owner")
-    repo = run_command(f"nix eval --raw -f {nixpkgs_dir} yaziPlugins.\"{plugin_name}\".src.repo")
+    owner = run_command(f'nix eval --impure --raw -f {nixpkgs_dir} "yaziPlugins.{plugin_name}.src.owner"')
+    repo = run_command(f'nix eval --impure --raw -f {nixpkgs_dir} "yaziPlugins.{plugin_name}.src.repo"')
```
Why is `--impure` required now?

GaetanLepage reviewed (commented):
I would recommend to split your changes in more commits.
The current ones mix cosmetic changes, with random refactorings and actual feature implementation.

khaneliman committed:
yaziPlugins.update: cleanup

Don't think we need to be so explicit on a lot of this. Also don't want
to support env variable for plugin update.

Signed-off-by: Austin Horstman <khaneliman12@gmail.com>

khaneliman committed:
yaziPlugins: update on 2025-06-06

- mediainfo: 25.5.28-unstable-2025-05-30 → 25.5.31-unstable-2025-06-05
- projects: 0-unstable-2025-05-29 → 0-unstable-2025-06-03
- relative-motions: 25.4.8-unstable-2025-04-16 → 25.5.28-unstable-2025-06-05
- restore: 25.5.28-unstable-2025-05-30 → 25.5.31-unstable-2025-06-05
- rich-preview: 0-unstable-2025-05-30 → 0-unstable-2025-05-31
- starship: 25.4.8-unstable-2025-05-30 → 25.4.8-unstable-2025-06-01

Signed-off-by: Austin Horstman <khaneliman12@gmail.com>

khaneliman committed:
yaziPlugins.rsync: update from 0-unstable-2025-04-24 to 0-unstable-2025-06-07

Signed-off-by: Austin Horstman <khaneliman12@gmail.com>

GaetanLepage reviewed (approved):
Looks good, nice work!

GaetanLepage commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 414418`
Commit: `23aefe4694ef2aa0902a6c7e8407b51c6e4f1224`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.rsync</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.rsync</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.rsync</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>yaziPlugins.mediainfo</li>
    <li>yaziPlugins.projects</li>
    <li>yaziPlugins.relative-motions</li>
    <li>yaziPlugins.rich-preview</li>
    <li>yaziPlugins.rsync</li>
    <li>yaziPlugins.starship</li>
  </ul>
</details>
